packages/eslint-plugin/src/rules/unified-signatures.ts

```diff
@@ -521,11 +532,10 @@ export default util.createRule({
         createScope(node.body, node.typeParameters);
       },
       TSTypeLiteral: createScope,
+
       // collect overloads
       TSDeclareFunction(node): void {
-        if (node.id && !node.body) {
```

Comment:
@bradzacher We're guaranteed that the `id` property will exist for a `TSDeclareFunction` based on TS's parsing.
However, the `body` property _may_ exist, but TS will emit an error during type checking (though it is a "grammar" error, but I'm not sure what that means). 

Given that, I thought it was safe to remove the condition. However, if you think that we should handle cases that already cause checking errors in TS, then I can add it back and add the non-compiling test. What's our philosophy here?

Reply:
The thing I'd be double checking to make sure is if you run linting on `declare function ()`, that eslint doesn't allow a "broken parse", and run the rules.

if that's not what happens, then I'm happy to remove the unnecessary check! Also in this case, if we know that it's guaranteed non-null, we might as well update the types so that everyone else doesn't do unnecessary checks.

Reply:
Verified that if there's a parse error, rules won't be run.